cloud/scope/powervs_machine_test.go

```diff
@@ -122,74 +133,1186 @@ func newDHCPServer(serverID, networkID string) models.DHCPServers {
+	})
+}
+
+func TestGetIgnitionVersion(t *testing.T) {
+	testcases := []struct {
+		name                    string
+		expectedIgnitionVersion string
+		scope                   PowerVSMachineScope
+	}{
+		{
+			name:                    "Ignition version is nil",
+			expectedIgnitionVersion: infrav1beta2.DefaultIgnitionVersion,
+			scope: PowerVSMachineScope{
+				IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{},
+			},
+		}, {
+			name:                    "Custom Ignition Version is set",
+			expectedIgnitionVersion: "3.4",
+			scope: PowerVSMachineScope{
+				IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+					Spec: infrav1beta2.IBMPowerVSClusterSpec{
+						Ignition: &infrav1beta2.Ignition{
+							Version: "3.4",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.name, func(_ *testing.T) {
+			g := NewWithT(t)
+			g.Expect(getIgnitionVersion(&tc.scope)).To(Equal(tc.expectedIgnitionVersion))
+		})
+	}
+}
+
+func TestBootstrapDataKey(t *testing.T) {
+	testcases := []struct {
+		name                     string
+		machineLabel             string
+		machineName              string
+		expectedBootstrapDataKey string
+	}{
+		{
+			name:                     "Returns BootstrapDataKey for a machine in control plane",
+			machineLabel:             capiv1beta1.MachineControlPlaneLabel,
+			machineName:              "foo-machine-0",
+			expectedBootstrapDataKey: path.Join("control-plane", "foo-machine-0"),
+		},
+		{
+			name:                     "Returns BootstrapDataKey for a worker node",
+			machineName:              "foo-machine-1",
+			machineLabel:             "foo",
+			expectedBootstrapDataKey: path.Join("node", "foo-machine-1"),
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.name, func(_ *testing.T) {
+			g := NewWithT(t)
+			machineScope := PowerVSMachineScope{
+				IBMPowerVSMachine: &infrav1beta2.IBMPowerVSMachine{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: tc.machineName,
+					},
+				},
+				Machine: &capiv1beta1.Machine{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: map[string]string{
+							tc.machineLabel: "",
+						},
+					},
+				},
+			}
+			g.Expect(tc.expectedBootstrapDataKey).To(Equal(machineScope.bootstrapDataKey()))
+		})
+	}
+}
+
+func TestGetNetworkID(t *testing.T) {
+	var (
+		mockCtrl    *gomock.Controller
+		mockpowervs *mock.MockPowerVS
+	)
+
+	setup := func(t *testing.T) {
+		t.Helper()
+		mockCtrl = gomock.NewController(t)
+		mockpowervs = mock.NewMockPowerVS(mockCtrl)
+	}
+	teardown := func() {
+		mockCtrl.Finish()
+	}
+
+	const networkID = "foo-network-id"
+	t.Run("Get Network ID", func(t *testing.T) {
+		t.Run("Returns networkID from Network spec's ID", func(t *testing.T) {
+			g := NewWithT(t)
+			scope := PowerVSMachineScope{}
+			expectedNetworkID := networkID
+			networkResource := infrav1beta2.IBMPowerVSResourceReference{
+				ID: core.StringPtr(expectedNetworkID),
+			}
+			result, err := getNetworkID(networkResource, &scope)
```

Comment:
```suggestion
			networkID, err := getNetworkID(networkResource, &scope)
```
Please change wherever applicable